federation/pkg/dnsprovider/providers/aws/route53/zones.go

```diff
@@ -34,15 +34,12 @@ func (zones Zones) List() ([]dnsprovider.Zone, error) {
 	var zoneList []dnsprovider.Zone
 
 	input := route53.ListHostedZonesInput{}
-	err := zones.interface_.service.ListHostedZonesPages(&input, func(page *route53.ListHostedZonesOutput, lastPage bool) bool {
+	zones.interface_.service.ListHostedZonesPages(&input, func(page *route53.ListHostedZonesOutput, lastPage bool) bool {
```

Comment:
You can't just ignore the error here.  If the route53 API returns an error to indicate that no zones exist, you need to return an empty list and no error.  But for other errors (for example authentication errors, or unreachable API) you need to return that error.